crates/typst-library/src/math/matrix.rs

```diff
@@ -200,27 +202,34 @@ impl LayoutMath for MatElem {
         // validate inputs
 
         let augment = self.augment(ctx.styles());
+        let rows = self.rows();
 
         if let Some(aug) = &augment {
             for &offset in &aug.hline.0 {
-                if offset == 0 || offset >= self.rows().len() {
+                if offset == 0
+                    || offset >= (rows.len() as isize)
+                    || offset <= -(rows.len() as isize)
```

Comment:
This (and the same thing below) could be
```rs
if offset == 0 || offset.unsigned_abs() >= rows.len()
```

Reply:
Good catch, thanks!